src/Lumper.UI/Services/BspService.cs

```diff
@@ -275,6 +278,12 @@ public async Task<bool> Save(IStorageFile? outFile)
             if (Program.Desktop.MainWindow is not null)
             {
                 var outName = outFile is not null ? Path.GetFileName(outFile.Path.AbsolutePath) : FileName;
+                if (outName != FileName)
+                {
+                    BspFile.GetLump<PakfileLump>().RenameDefaultCubemap(outName);
+                    BspService.Instance.PakfileLumpViewModel.IsModified = true;
+                    BspService.Instance.ResetLumpViewModels();
+                }
```

Comment:
Still don't think this is the right approach, should do this logic in the BspFile SaveToFile method. Probably worth adding a checkbox to the File... menu to toggle this in UI though, then pass that into SaveToFile.

I'd like to avoid reloading lump viewmodels here, since that would cause every UI page to reload. Unfortunately I can't think of a way to update the viewmodels that isn't ugly, but still worth it to avoid reloading the whole things. After SaveToFile is called, if successful, iterate through every PakfileEntryViewModel and update their Key to the Key of the underlying PakfileEntry, if changed.

Don't love doing this but should be perfectly performant, and will work if there's ever other pakfile changes we make during saving in the future.